07a_system_activity_plots.R

```diff
@@ -269,44 +268,70 @@ ggplot(df, aes(x = group.id, fill = Status)) +
 syso_detailBox <- reactive({
   # remove group names from race/ethnicity filter
   # so we can use getNameByValue() to grab the selected option label
-  syso_race_ethnicities <- unlist(syso_race_ethnicity_cats())
-  names(syso_race_ethnicities) <- gsub("Group [0-9]+\\.", "",
-                                       names(syso_race_ethnicities))
-  
+  # if (input$methodology_type == 2) {
+    # browser()
+  # }
+
   list(
```

Comment:
Can we clean up this whole block? It's a little hard to see what's going on. Something like this?

```
  detail_line <- function(detail_label, val_list, inputVal) {
    return(
      HTML(glue(
        "<b>{detail_label}:</b> {getNameByValue(val_list, inputVal)} <br>"
      ))
    )
  }
  list(
    strong("Date Range: "),
    
    ReportStart(), " to ", ReportEnd(), br(),
    
    if (getNameByValue(syso_hh_types, input$syso_hh_type) != "All People")
      detail_line("Household Type", syso_hh_types, input$syso_hh_type),
    
    detail_line("Level of Detail", syso_level_of_detail, input$syso_level_of_detail),
    
    if (getNameByValue(syso_project_types, input$syso_project_type) != "All")
      detail_line("Project Type", syso_project_types, input$syso_project_type),
    
    detail_line("Methodology Type", syso_methodology_types, input$methodology_type),

    if (length(input$syso_age) != length(syso_age_cats))
      HTML(glue(
        "<b>Age:</b> {paste(input$syso_age, collapse = ', ')} <br>"
      )),
    
    if (length(input$syso_gender) != length(syso_gender_cats(input$methodology_type)))
      detail_line("Gender", syso_gender_cats(input$methodology_type), input$syso_gender),
    
    if (getNameByValue(syso_race_ethnicity_cats(input$methodology_type),
                       input$syso_race_ethnicity) != "All Races/Ethnicities")
      strong("Race/Ethnicity: "), str_sub(
        getNameByValue(
          unlist(syso_race_ethnicity_cats(input$methodology_type)),
          input$syso_race_ethnicity
        ),
        start = str_locate(getNameByValue(
          unlist(syso_race_ethnicity_cats(input$methodology_type)),
          input$syso_race_ethnicity
        ), "\\.")[, 1] + 1,
        end = -1L
      ), br(),
    if(getNameByValue(syso_spec_pops_people, input$syso_spec_pops) != "None")
      detail_line("Special Populations", syso_spec_pops_people, input$syso_spec_pops)
```